snuba/query/allocation_policies/errors_allocation_policy.py

```diff
@@ -0,0 +1,186 @@
+        "migration.backfill_perf_issue_events_issue_platform",
+        "api.vroom",
+        "replays.query.download_replay_segments",
+        "outcomes.timeseries",
+        "release_monitor.fetch_projects_with_recent_sessions",
+        "https://snuba-admin.getsentry.net/",
+        "reprocessing2.start_group_reprocessing",
+    ]
+)
+
+
+class ErrorsAllocationPolicy(AllocationPolicy):
+
+    WINDOW_SECONDS = 10 * 60
+    WINDOW_GRANULARITY_SECONDS = 60
+
+    def __init__(
+        self,
+        storage_set_key: StorageSetKey,
+        required_tenant_types: list[str],
+        **kwargs: str,
+    ) -> None:
+        super().__init__(storage_set_key, required_tenant_types)
+        self._rate_limiter = RedisSlidingWindowRateLimiter()
+
+    @property
+    def rate_limit_prefix(self) -> str:
+        return self.__class__.__name__
+
+    def _are_tenant_ids_valid(
+        self, tenant_ids: dict[str, str | int]
+    ) -> tuple[bool, str]:
+        if "referrer" not in tenant_ids:
+            return False, "no referrer"
+        if (
+            "organization_id" not in tenant_ids
+            and tenant_ids.get("referrer", None) not in _ORG_LESS_REFERRERS
+        ):
+            return False, "no organization_id for referrer {tenant_ids['referrer']}"
+        return True, ""
+
+    def _get_quota_allowance(self, tenant_ids: dict[str, str | int]) -> QuotaAllowance:
+        # TODO: This kind of killswitch should just be included with every allocation policy
+        is_active = get_config(f"{self.rate_limit_prefix}.is_active", True)
+        is_enforced = get_config(f"{self.rate_limit_prefix}.is_enforced", False)
+        if not is_active:
+            return DEFAULT_PASSTHROUGH_POLICY.get_quota_allowance(tenant_ids)
+        ids_are_valid, why = self._are_tenant_ids_valid(tenant_ids)
+        if not ids_are_valid and is_enforced:
+            metrics.increment(
+                "db_request_rejected",
+                tags={
+                    "storage_set_key": self._storage_set_key.value,
+                    "is_enforced": str(is_enforced),
+                },
+            )
+            return QuotaAllowance(
+                can_run=False, max_threads=0, explanation={"reason": why}
+            )
+        if "organization_id" in tenant_ids:
+            org_scan_limit = get_config(
+                # TODO: figure out an actually good number
+                f"{self.rate_limit_prefix}.org_scan_limit",
+                10000,
+            )
+
+            timestamp, granted_quotas = self._rate_limiter.check_within_quotas(
+                [
+                    RequestedQuota(
+                        self.rate_limit_prefix,
+                        # request a big number because we don't know how much we actually
+                        # will use in this query. this doesn't use up any quota, we just want to know how much is left
+                        int(1e10),
```

Comment:
What are the units for this value? Is it bytes scanned?

Reply:
good question. Yes but I will clarify that